eclipse-collections-code-generator/src/main/resources/api/set/immutablePrimitiveSet.stg

```diff
@@ -43,6 +43,16 @@ public interface Immutable<name>Set extends Immutable<name>Collection, <name>Set
     @Override
     \<V> ImmutableSet\<V> collect(<name>ToObjectFunction\<? extends V> function);
 
+    /**
+     * @since 11.0.
+     */
+    @Override
+    default Immutable<name>Set union(<name>Set set)
+    {
+        Mutable<name>Set newSet = this.toSet();
```

Comment:
use the larger of the two sets as starting point.

Reply:
Updated.